pip/req/req_set.py

```diff
@@ -570,13 +602,32 @@ def _pip_has_created_build_dir(self):
             )
         )
 
+    def _to_install(self):
+        """Create the installation order.
+
+        We install the user specified things in the order given, except when
+        dependencies require putting other things first.
```

Comment:
this doc string implies we *have* to sort this way, but we really don't.   we're doing this mostly for the case of packages that don't have setup_requires done properly?  or... is there another reason?